example-library/src/main/java/uk/gov/gchq/palisade/example/config/ExampleUserCacheWarmerFactory.java

```diff
@@ -0,0 +1,158 @@
+/*
+ * Copyright 2020 Crown Copyright
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.gchq.palisade.example.config;
+
+import org.springframework.boot.context.properties.ConfigurationProperties;
+
+import uk.gov.gchq.palisade.Generated;
+import uk.gov.gchq.palisade.User;
+import uk.gov.gchq.palisade.example.common.ExampleUser;
+import uk.gov.gchq.palisade.example.common.TrainingCourse;
+import uk.gov.gchq.palisade.service.UserCacheWarmerFactory;
+
+import java.util.Collections;
+import java.util.EnumSet;
+import java.util.Objects;
+import java.util.Set;
+import java.util.StringJoiner;
+
+import static java.util.Objects.requireNonNull;
+
+@ConfigurationProperties
+public class ExampleUserCacheWarmerFactory implements UserCacheWarmerFactory {
+
+    private String userId;
+    private Set<String> auths;
+    private Set<String> roles;
+    private EnumSet<TrainingCourse> trainingCourses;
+
+    /**
+     * Constructor with 0 arguments for an example implementation
+     * of the {@link UserCacheWarmerFactory} interface
+     */
+    public ExampleUserCacheWarmerFactory() {
```

Comment:
Ditto previous comments, empty constructor and values at initialisation

Reply:
Same action performed as the previous comment.